### PR TITLE
feat: add moose harness init

### PR DESCRIPTION
**Pull request by callicles**
> [!NOTE]
> **Medium Risk**
> Medium risk because it introduces a new CLI flow that writes to user/editor config files, installs global tooling via `npm`, and can fetch and install content from GitHub, which increases surface area for failures and unintended local changes.
> 
> **Overview**
> **Adds `moose harness init`** to scaffold a new Moose project *and* configure a developer harness in the same run, with a zero-arg interactive wizard and a non-interactive arg/JSON-driven mode (including `schema --json` for a versioned input contract).
> 
> The harness setup can auto-detect or explicitly target coding agents, install and symlink skills into agent-specific directories, and configure MCP (and optionally MooseStack LSP) by updating the appropriate JSON/TOML/YAML user config files with backups.
> 
> Refactors existing `init` logic by extracting project scaffolding + optional remote ClickHouse bootstrap into `initialize_project`, and updates docs/templates plus adds Rust + E2E tests and fixtures covering wizard flow, CLI precedence, config output, and edge cases like `--name schema` and missing `--from-remote` value.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit dbeb39eed0d3a80a46849fd80e7b8ce0a3161734. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by ghost** [comment]
> **Environment:** `preview branch: (nico/eng-2543-pr-comment-should-explain-when-a-filtered-push-does-not)`

The latest updates on your projects. [Learn more about Boreal](https://www.boreal.cloud).


> **Note:** No new deployment was triggered for `dbeb39e`. The pushed files did not match any project's root path or the shared `packages/` directory.

<details>
<summary>Skipped projects (no relevant changes)</summary>

- **moosestack** - no changes in `/examples/fastify-moose/moose`

</details>

**Comment by callicles** [comment]
* **#3868** <a href="https://app.graphite.com/github/pr/514-labs/moosestack/3868?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/514-labs/moosestack/3868?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: apps/framework-cli/src/cli/routines/harness/config.rs [hunk 1]
```diff
@@ -0,0 +1,965 @@
...
+    #[error("Command `{command}` timed out after {seconds}s")]
+    CommandTimeout { command: String, seconds: u64 },
+}
+
+pub(crate) async fn install_lsp() -> Result<bool, HarnessConfigError> {
+    if !command_exists("npm") {
+        show_message!(
+            MessageType::Warning,
+            Message::new(
+                "Harness".to_string(),
```
Comment (georgevanderson) [inline apps/framework-cli/src/cli/routines/harness/config.rs]: theres 65 of these "Harness".to_string() in this file, and more elsewhere. constant?

File: apps/framework-cli/src/cli/routines/harness/config.rs [hunk 2]
```diff
@@ -0,0 +1,965 @@
...
+            McpConfig::YamlFile { target } => {
+                let mut wrote_any = false;
+                for path in target.resolve_paths(home).map_err(|error| {
+                    HarnessConfigError::ConfigTargetResolution(error.to_string())
+                })? {
+                    wrote_any = write_mcp_yaml_to_path(&path)? || wrote_any;
+                }
+                wrote_any
+            }
+        };
```
Comment (georgevanderson) [inline apps/framework-cli/src/cli/routines/harness/config.rs:193]: why does JSON need all written but toml and yaml are ||?

File: apps/framework-cli/src/cli/routines/harness/mod.rs [hunk 3]
```diff
@@ -0,0 +1,2012 @@
...
+                "Harness".to_string(),
+                format!(
+                    "Failed to replace canonical skill directory {}",
+                    canonical_dir.display()
+                ),
+            ),
+            e,
+        )
+    })?;
+    std::mem::forget(staging_dir);
```
Comment (georgevanderson) [inline apps/framework-cli/src/cli/routines/harness/mod.rs]: I just tried locally removing this. alternative was `let _ = staging_dir.into_path();` looks gross but no leak. up to you

File: apps/framework-cli/src/cli/routines/harness/config.rs (hunk 2 above)
Comment (callicles) [reply apps/framework-cli/src/cli/routines/harness/config.rs:193]: The difference is intentional.

For JSON targets, `ConfigTarget::resolve_paths()` can expand to multiple files for a single logical agent install, especially VS Code user profiles. In that case I want the overall result to be `false` if any profile was skipped or required manual intervention, even if another profile was updated successfully. That behavior is covered by `test_write_agent_mcp_returns_false_if_any_vscode_profile_is_skipped`.

For TOML and YAML targets here, each config currently resolves to a single path, so `wrote_any` is enough and `wrote_any && all_written` would collapse to the same result. If we add multi-path TOML/YAML targets later, I’d make them follow the same aggregate semantics as the JSON branch.
